octoprint/docker-compose.yml

```diff
@@ -0,0 +1,17 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: octoprint_web_1
+      APP_PORT: 5000
+      PROXY_AUTH_ADD: "false"
+
+  web:
+    image: octoprint/octoprint:1.8.7
+    volumes:
+      - ${APP_DATA_DIR}/data:/data
+      - ${UMBREL_ROOT}/octoprint:/octoprint
+    devices:
+      - "/dev/ttyACM0:/dev/ttyACM0"
```

Comment:
I think this could cause us some issues. 

If there is no printer connected to `/dev/ttyACM0` on the Umbrel device then when the container is booting up during install/restart it will error out and not start up because it won't be able to map `dev/ttyACM0` from the container to a non-existent `/dev/ttyACM0` file on the Umbrel device.

We could potentially get around this by just mapping `- "/dev:/dev"` but I don't think we should give an app access to the entire `/dev` folder for security reasons.

Have you tested whether you can connect a printer with these two line commented out? Unfortunately I do not have access to a 3d printer to check.

Reply:
Yes, youre right. If the printer is not connected, the image will not boot. I'm trying to write the dir if is doesnt exists. Let me try some things.

Reply:
My workaround doesnt work :( i was inspired by this https://github.com/getumbrel/umbrel-apps/pull/43/files#diff-de62ede3db7fbfc30c6ec34ff877c60a6afad419a257528af36e1bc311d81a97

Do you find a solution for esphome ?

Reply:
![Captura desde 2023-03-24 20-13-23](https://user-images.githubusercontent.com/25598400/227659750-9be2b413-a0b3-42fb-8870-9f3eadc2b536.png)

When the printer is connected, it works really well, and once the image is booted, you could disconnect and connect it at any time. The problem is on installation and restart

Reply:
@nmfretz Is there a possible solution to handle this ?